src/applications/personalization/profile/util/index.js

```diff
@@ -110,39 +228,63 @@ export const isSignedUpForCNPDirectDeposit = apiData =>
 export const isSignedUpForEDUDirectDeposit = apiData =>
   !!eduDirectDepositAccountNumber(apiData);
 
+const getLighthouseErrorCode = (errors = []) => {
+  // there should only be one error code in the errors array, but just in case
+  const error = errors.find(err => err?.code);
+  return `${error?.code || OTHER_ERROR_GA_KEY} ${error?.detail || ''}`;
+};
+
+const getPPIUErrorCode = (errors = []) => {
+  if (hasAccountFlaggedError(errors)) {
+    return PPIU_ERROR_MAP.ACCOUNT_FLAGGED_FOR_FRAUD.GA_KEY;
+  }
+
+  if (hasRoutingNumberFlaggedError(errors)) {
+    return PPIU_ERROR_MAP.ROUTING_NUMBER_FLAGGED_FOR_FRAUD.GA_KEY;
+  }
+
+  if (hasInvalidRoutingNumberError(errors)) {
+    return PPIU_ERROR_MAP.INVALID_ROUTING_NUMBER.GA_KEY;
+  }
+
+  if (hasInvalidAddressError(errors)) {
+    return PPIU_ERROR_MAP.BAD_ADDRESS.GA_KEY;
+  }
+
+  if (hasInvalidHomePhoneNumberError(errors)) {
+    return PPIU_ERROR_MAP.BAD_HOME_PHONE.GA_KEY;
+  }
+
+  if (hasInvalidWorkPhoneNumberError(errors)) {
+    return PPIU_ERROR_MAP.BAD_WORK_PHONE.GA_KEY;
+  }
+
+  if (hasPaymentRestrictionIndicatorsError(errors)) {
+    return PPIU_ERROR_MAP.PAYMENT_RESTRICTIONS_PRESENT.GA_KEY;
+  }
+
+  return PPIU_ERROR_MAP.GENERIC_ERROR.GA_KEY;
+};
```

Comment:
Another long function that will be able to be removed once the move to the new endpoint is complete.